main.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+This program will help you find out how much your normal weight is. In case of discrepancy, help with advice.
+*/
+
+#include <iostream>
+#include <cmath>
+#include <string>
+
+using namespace std;
+
+enum BodyMassIndexes
+{
+    SEVERE_THINNESS = 16,
+    MODERATE_THINNESS = 17,
+    SLIGHT_THINNESS = 19,
+    NORMAL_BODY_WEIGHT = 25,
+    OVERWEIGHT = 30,
+    FIRST_LEVEL_OBESITY = 35,
+    SECOND_LEVEL_OBESITY = 40
+};
+
+void flush(istream& in)
+{
+    in.ignore(numeric_limits<streamsize>::max(), '\n');
+    in.clear();
+}
+
+void pause()
+{
+    cout << "Press [Enter] to continue . . ." << endl;
+    cin.get();
+}
+
+void advice_printer(double normalBodyWeigthMin, double normalBodyWeigthMax)
+{
+    cout << "Your normal body weight between " << normalBodyWeigthMin << " and " << normalBodyWeigthMax << " kg." << endl;
+    cout << "Increase your calorie intake until you reach the right weight." << endl;
+}
+
+void wrong_input_printer()
+{
+    cout << "You must have not correct input." << endl;
+    cout << "Restart and pay more attention!" << endl;
+}
+
+double height_square_calculator(double height)
+{
+    return pow((height / 100.00), 2);
+}
+
+double normal_body_weight_calculator(double bodyWeight, double normalBodyWeight)
+{
+    return bodyWeight - normalBodyWeight;
+}
+
+int main()
+{
+    cout << "This program will help you find out how much your normal weight is." << endl;
+    cout << "In case of discrepancy, help with advice!" << endl;
+    cout << "Just follow the instructions!" << endl;
+    cout << endl;
+
+    double bodyWeight = 0.0;
+    double height = 0.0;
+    double age = 0.0;
+    double reduceValue = 0.0;
+    int bodyMassIndex = 0;
+    double normalBodyWeigthMin = 0.0;
+    double normalBodyWeigthMax = 0.0;
+
+    cout << "Enter your body weight (kg): ";
+    cin >> bodyWeight;
+
+    if (bodyWeight <= 0 || bodyWeight > 350 || (isnan(bodyWeight) != 0)) {
+        wrong_input_printer();
+        return -1;
+    }
+
+    cout << "Enter your height (cm): ";
+    cin >> height;
+
+    if (height <= 0 || height > 250 || (isnan(height) != 0)) {
+        wrong_input_printer();
+        return -1;
+    }
+    
+    cout << "Enter your age: ";
+    cin >> age;
+
+    if (age < 0 || age > 130 || (isnan(age) != 0)) {
+        wrong_input_printer();
+        return -1;
+    }
+
+    cout << endl;
+
+    bodyMassIndex = static_cast<int>(bodyWeight / height_square_calculator(height));
+    normalBodyWeigthMin = SLIGHT_THINNESS * height_square_calculator(height);
+    normalBodyWeigthMax = NORMAL_BODY_WEIGHT * height_square_calculator(height);
+
+    if (bodyMassIndex < SEVERE_THINNESS) {
+        cout << "You are seriously skinny!" << endl;
+        advice_printer(normalBodyWeigthMin, normalBodyWeigthMax);
+
+        return 0;
+    }
+    else if (bodyMassIndex < MODERATE_THINNESS && bodyMassIndex >= SEVERE_THINNESS) {

```

Comment:
nem kell a 2. feltétel, bodymassIndex biztos >= SEVERE_THINNESS mert az előző if-be nem mentünk bele.